### DISCUSSION TITLE
Tecnologia para Backend

### DISCUSSION THREAD

**Discussion by timeuz** [frontendbr/forum#2143] [Perguntas e Respostas]
Pessoas, uns meses atrás, como estudo, desenvolvi um sisteminha em PHP para minha esposa cadastrar os jobs que ela fazia e o sisteminha calcula o valor de cada job, verifica e exibe quanto falta para bater a meta do mês e gera relatórios mensais.
O problema é que fiz em procedural e estou estudando JavaScript OO agora. Andei lendo que acessar banco de dados com JS não é nada seguro e como pretendo distribuir esse sisteminha 2.0 OO gostaria de saber o que vocês recomendam para fazer essa parte de interação com o banco de dados.
PHP já que tenho uma pequena experiência? Node já que é JavaScript? Outro?

**Comment by victorferreira** [comment]
> Andei lendo que acessar banco de dados com JS não é nada seguro

Isso não é verdade. Qualquer linguagem que você escolha pra escrever o backend deve cumprir bem o propósito. Uma das grandes vantagens do Node é poder usar a mesma linguagem tanto no backend quanto no frontend, mas PHP vai servir bem também.

**Comment by wilsonneto-swe** [comment]
@heathz não sei onde você leu isso, mas para de ler coisas desta fonte ai rs

JS, .Net, PHP mesmo você estará muito bem servido. Vai de gosto e de demanda, mas estas demandas mais genericas, são tranquilas em qualquer linguagem.

Forte abraço!

**Comment by igorlvicente** [comment]
Talvez você leu que acessar o banco de dados pelo frontend (ou seja, Javascript) é inseguro. Isso não quer dizer que é inseguro acessar com Javascript pelo backend, como os colegas acima disseram.

**Reply by timeuz** [reply]
Então... talvez eu tenha me expressado mal. Era isso mesmo que eu estava querendo dizer hehehe... 
E a dúvida em si era justamente sobre usar Javascript para fazer o backend, se era uma boa opção ou se seria melhor utilizar alguma outra linguagem levando em consideração segurança, desempenho e curva de aprendizado.
Meu foco é frontend mas sinto a necessidade de saber mais sobre backend principalmente para coisas pequenas que preciso no dia-a-dia.

**Reply by igorlvicente** [reply]
A minha dica é para seguir o que o coração manda. Faz na linguagem que tens vontade de fazer. Os benefícios da vontade superarão os possíveis malefícios que terás caso escolha uma linguagem "ruim".
